### PR TITLE
fix: UI issues with code editor

### PR DESCRIPTION
**Pull request by arunvjn**
## Description
https://github.com/appsmithorg/appsmith/pull/22652 introduced new styles to code in Appsmith. Along with it was some unintentional changes. 
Bundle optimization [PR](https://github.com/appsmithorg/appsmith/pull/21667) introduced, CodeEditorFallback component which incorrectly renders borders and causes UI to flicker. It also broke styles of lint message popover. This PR fixes these issues.

Code changes
- Added 6px padding to code editor lines
- Use class variables instead of ref
- Removed border for JSEditor's CodeEditorFallback component
- Fixes lint message style
>
> Links to Notion, Figma or any other documents that might be relevant to the PR
> https://www.notion.so/appsmith/Visual-changes-to-code-in-Appsmith-8bb530c60ee844e5b44adec039bf9280#d99e8450f022439f845b5a0d7deb1d3f
>
#### PR fixes following issue(s)
Fixes # (issue number)
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
#### Type of change
- Bug fix (non-breaking change which fixes an issue)
>
>
>
#### How Has This Been Tested?
- [x] Manual
>
>
#### Test Plan
Visual checks to test for changes as per [this](https://www.notion.so/appsmith/Error-tooltip-for-data-fields-on-property-pane-Needs-design-810cdc489c3e4168a0b1b348730ac98c?utm_content=810cdc48-9c3e-4168-a0b1-b348730ac98c&utm_campaign=TGA9SH07N&n=slack&n=slack_link_unfurl&pvs=6) Notion doc
>
#### Issues raised during DP testing
None
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by ramsaptami** [comment]
lgtm cc: @arunvjn

File: app/client/src/globalStyles/CodemirrorHintStyles.ts [hunk 1]
```diff
@@ -306,4 +257,56 @@ export const CodemirrorHintStyles = createGlobalStyle<{
...
+        props.editorTheme === EditorTheme.DARK
+          ? "#23292e"
+          : "#DEDEDE"} !important;
+      box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.12) !important;
+      overflow: scroll;
+    }
+  }
+  .CodeMirror-lint-tooltip {
+    && {
+      border: none;
```
Comment (ravikp7) [inline app/client/src/globalStyles/CodemirrorHintStyles.ts:285]: @arunvjn I see this change that you've added `&& {` in the styles of three classes here. What does this change do?

File: app/client/src/components/editorComponents/ReadOnlyEditor.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ function ReadOnlyEditor(props: Props) {
       : true,
     borderLess: true,
     folding: props.folding,
-    isReadOnly: props.isReadOnly,
+    isReadOnly: true,
```
Comment (ravikp7) [inline app/client/src/components/editorComponents/ReadOnlyEditor.tsx:42]: Let's remove it from props as well?

File: app/client/src/components/editorComponents/CodeEditor/styledComponents.ts [hunk 3]
```diff
@@ -129,6 +129,7 @@ export const EditorWrapper = styled.div<{
       border-radius: 0px;
       font-family: ${(props) => props.theme.fonts.code};
       font-size: 13px;
+      ${(props) => props.isReadOnly && "font-size: 12px"};
```
Comment (ravikp7) [inline app/client/src/components/editorComponents/CodeEditor/styledComponents.ts:132]:
```suggestion
      font-size:  ${(props) => props.isReadOnly ? "12px" : "13px"};
```

File: app/client/src/components/editorComponents/CodeEditor/index.tsx [hunk 4]
```diff
@@ -278,8 +278,8 @@ class CodeEditor extends Component<Props, State> {
   annotations: Annotation[] = [];
   updateLintingCallback: UpdateLintingCallback | undefined;
   private editorWrapperRef = React.createRef<HTMLDivElement>();
+  currentLineNumber: number | null = null;
```
Comment (ravikp7) [inline app/client/src/components/editorComponents/CodeEditor/index.tsx:281]: Noice!

File: app/client/src/components/editorComponents/ReadOnlyEditor.tsx (hunk 2 above)
Comment (arunvjn) [reply app/client/src/components/editorComponents/ReadOnlyEditor.tsx:41]: Done

File: app/client/src/components/editorComponents/LazyCodeEditor/styles.tsx [hunk 5]
```diff
@@ -3,34 +3,41 @@ import { ContentKind } from "./types";
...
+    }};
 
     word-wrap: break-word !important;
     white-space: pre-wrap !important;
     word-break: normal !important;
 
     color: ${({ contentKind }) =>
       contentKind === ContentKind.CODE
-        ? "#063289"
+        ? "#304eaa"
```
[the remark this answers is not in the document]
Comment (arunvjn) [reply app/client/src/components/editorComponents/LazyCodeEditor/styles.tsx]: Done

**Review by ravikp7 — APPROVED** [review]

**Pull request merged**